e3sm_diags/driver/aerosol_aeronet_driver.py

```diff
@@ -25,74 +27,110 @@
-        test_data = utils.dataset.Dataset(parameter, test=True)
-        parameter.test_name_yrs = utils.general.get_name_and_yrs(
-            parameter, test_data, season
-        )
-        parameter.ref_name_yrs = "AERONET (2006-2015)"
+    test_ds = Dataset(parameter, data_type="test")
 
-        for var in variables:
-            logger.info("Variable: {}".format(var))
-            parameter.var_id = var
+    for var_key in variables:
+        logger.info("Variable: {}".format(var_key))
+        parameter.var_id = var_key
 
-            test = test_data.get_climo_variable(var, season)
-            test_site = interpolate_model_output_to_obs_sites(test, var)
+        for season in seasons:
+            ds_test = test_ds.get_climo_dataset(var_key, season)
+            da_test = ds_test[var_key]
 
-        if run_type == "model_vs_model":
-            ref_data = utils.dataset.Dataset(parameter, ref=True)
-            parameter.ref_name_yrs = utils.general.get_name_and_yrs(
-                parameter, ref_data, season
+            test_site_arr = interpolate_model_output_to_obs_sites(
+                ds_test[var_key], var_key
             )
-            ref = ref_data.get_climo_variable(var, season)
-            ref_site = interpolate_model_output_to_obs_sites(ref, var)
 
-        elif run_type == "model_vs_obs":
-            ref_site = interpolate_model_output_to_obs_sites(None, var)
-        else:
-            raise ValueError("Invalid run_type={}".format(run_type))
+            parameter.test_name_yrs = test_ds.get_name_yrs_attr(season)
+            parameter.ref_name_yrs = "AERONET (2006-2015)"
 
-        parameter.output_file = (
-            f"{parameter.ref_name}-{parameter.var_id}-{season}-global"
-        )
-        aerosol_aeronet_plot.plot(test, test_site, ref_site, parameter)
+            if run_type == "model_vs_model":
+                ref_ds = Dataset(parameter, data_type="ref")
+
+                parameter.ref_name_yrs = utils.general.get_name_and_yrs(
+                    parameter, ref_ds, season
+                )
+
+                ds_ref = ref_ds.get_climo_dataset(var_key, season)
+                ref_site_arr = interpolate_model_output_to_obs_sites(
+                    ds_ref[var_key], var_key
+                )
+            elif run_type == "model_vs_obs":
+                ref_site_arr = interpolate_model_output_to_obs_sites(None, var_key)
+            else:
+                raise ValueError("Invalid run_type={}".format(run_type))
+
+            parameter.output_file = (
+                f"{parameter.ref_name}-{parameter.var_id}-{season}-global"
+            )
+
+            metrics_dict = {
+                "max": da_test.max().item(),
+                "min": da_test.min().item(),
+                "mean": spatial_avg(ds_test, var_key, axis=["X", "Y"]),
+            }
+            aerosol_aeronet_plot.plot(
+                parameter, da_test, test_site_arr, ref_site_arr, metrics_dict
+            )
```

Comment:
For consistency across refactored sets, I moved the metrics dictionary out of the plot function and into the driver.